.github/workflows/build-examples.yml

```diff
@@ -0,0 +1,57 @@
+name: Build Library Examples
+on: [push, pull_request]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        fqbn: [
+          "arduino:samd:mkrwifi1010",
+          "arduino:samd:nano_33_iot",
+          "arduino:megaavr:uno2018:mode=on",
+          "arduino:mbed:nano33ble"
+        ]
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
+
+      - name: Setup arduino-cli
+        env:
+          VERSION: latest
+        run: |
+          set -x
+          wget -P $HOME https://downloads.arduino.cc/arduino-cli/arduino-cli_${VERSION}_Linux_64bit.tar.gz
+          mkdir $HOME/bin
+          tar xf $HOME/arduino-cli_${VERSION}_Linux_64bit.tar.gz  -C $HOME/bin
+          echo ::add-path::$HOME/bin
+
+      - name: Export FQBN and BOARD
+        run: |
+          FQBN=${{ matrix.fqbn }}
+          CORE=`echo "$FQBN" | cut -d':' -f1,2`
+          echo ::set-env name=FQBN::"$FQBN"
```

Comment:
@sandeepmistry May I ask also for a small comment explaining each step? In this line it took me a couple of minutes before realizing this is a simple message echo, instead of an obscure bash magic to export the variable :) 

Also in https://github.com/arduino-libraries/ArduinoBLE/pull/31/files#diff-3ce900fec73e52de9493561cd47d3a5bR44 would be nice to say why we symlink the folder instead of, for example, moving the library into my sketchbook or something similar.

WDYT?

Reply:
Sure thing, will add some comments :)

> simple message echo, instead of an obscure bash magic to export the variable :)

it is magic! It's the trick to export env vars to steps: https://help.github.com/en/articles/development-tools-for-github-actions#set-an-environment-variable-set-env